packages/passport/sdk/src/react/PassportContext.tsx

```diff
@@ -0,0 +1,242 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import React, {
+  createContext, useContext, useEffect, useMemo,
+  useState,
+} from 'react';
+import { Web3Provider } from '@ethersproject/providers';
+import { Passport } from '../Passport';
+import { PassportModuleConfiguration, UserProfile } from '../types';
+
+type PassportContextType = {
+  passportInstance: Passport;
+  isLoggedIn: boolean;
+  login: () => Promise<string[]>;
+  logout: () => Promise<void>;
+  loginWithoutWallet: () => Promise<UserProfile | null>;
+  loginWithEthersjs: () => Promise<{ accounts: string[], provider: Web3Provider | null }>;
```

Comment:
When would provider be null here?

Reply:
I'll move provider onto the context but it's null when `provider.send('eth_requestAccounts', [])` rejects.